src/core/dash_internal.h

```diff
@@ -1506,46 +1506,22 @@ template <bool UV>
 std::enable_if_t<UV, unsigned> Segment<Key, Value, Policy>::CVCOnBump(uint64_t ver_threshold,
                                                                       unsigned bid, unsigned slot,
                                                                       Hash_t hash,
-                                                                      uint8_t result_bid[2]) const {
+                                                                      uint8_t result_bid[3]) const {
   if (bid < kRegularBucketCnt) {
     // right now we do not migrate entries from nid to bid, only from stash to normal buckets.
     return 0;
   }
 
   // Stash case.
+  unsigned result = 0;
   if (bucket_[bid].GetVersion() < ver_threshold) {
-    // if we move an entry from a bucket that has lower version
```

Comment:
can you please explain the bug in the PR description. Also please add comments here, similarly to how I tried to voice the logic in the comments are deleted.
Finally, a dash_test is missing. It should be pretty easy to add a unit test if you know the scenario for which the bug exists.

Reply:
+1 and I added a bunch of comments. I think now it's a little bit clearer. 

For the unit we already have it, but we expected the wrong results (see the changes and the comment on the test). Keep in mind that the actual failure was caused during replication (and the reg test changes), because we would miss (not send) one of the buckets that got modified (and whose version was less than the threshold). This caused inconsistent data between the master and replica

we already have the one I introduced (the regression which failed)

Reply:
thanks!